### PR TITLE
Basic i18n_string integration.

### PR DESCRIPTION
**Pull request by adam-vessey**
Allows for translation of the field labels.

### CODE REVIEW

File: includes/admin.inc [hunk 1]
```diff
@@ -1361,20 +1372,26 @@ function _islandora_solr_handle_solr_field_settings($solr_field_settings = NULL,
...
-          'range_facet_start' => $solr_field_settings['range_facet_start'],
-          'range_facet_end' => $solr_field_settings['range_facet_end'],
-          'range_facet_gap' => $solr_field_settings['range_facet_gap'],
-          'date_facet_format' => $solr_field_settings['date_facet_format'],
-          'range_facet_slider_enabled' => $solr_field_settings['range_facet_slider_enabled'],
-          'range_facet_slider_color' => $solr_field_settings['range_facet_slider_color'],
-          'date_filter_datepicker_enabled' => $solr_field_settings['date_filter_datepicker_enabled'],
-          'date_filter_datepicker_range' => $solr_field_settings['date_filter_datepicker_range'],
-          'permissions' => $solr_field_settings['permissions'],
-        );
```
Comment (adam-vessey) [inline includes/admin.inc:1377]: Was generating a large number of strict warnings due to trying to access things blindly, which caused the AJAX request to fail... Great fun.

File: includes/admin.inc [hunk 2]
```diff
@@ -1301,7 +1311,8 @@ function _islandora_solr_admin_settings_field_configure($form, &$form_state) {
   $title = t('Configure field:') . ' ' . $solr_field;
   $commands[] = ajax_command_invoke('#islandora-solr-admin-dialog-title', 'html', array($title));
   // dialog body
-  $body = drupal_render(drupal_get_form($form_function, $variables));
```
Comment (adam-vessey) [inline includes/admin.inc:1304]: Strict warning: drupal_render() takes a reference.

File: includes/db.inc [hunk 3]
```diff
@@ -42,22 +42,28 @@ function islandora_solr_get_fields($field_type = NULL, $filter = TRUE, $simplifi
...
+    islandora_solr_translate_field_labels($records);
+  }
+
   // simplify
   if ($simplified == TRUE) {
     $records = _islandora_solr_simplify_fields($records);
   }
   // fields as keys
   if ($keys == TRUE && $simplified == FALSE) {
-    foreach ($records as $key => $value) {
```
Comment (adam-vessey) [inline includes/db.inc:58]: Unsetting while iterating is usually to be avoided... Made it so.

**Comment by philsogaDGI** [comment]
Everything looks good.  I installed and tested on my local machine.

**Pull request merged**